catalog/MDCCatalog/MDCNodeListViewController.swift

```diff
@@ -578,6 +578,9 @@ extension MDCNodeListViewController {
       vc = MDCNodeListViewController(node: node)
     }
     self.navigationController?.pushViewController(vc, animated: true)
+    let MDCAppBarNavigationController = self.navigationController as? MDCAppBarNavigationController
+    let appBar = MDCAppBarNavigationController?.appBarViewController(for: vc)
+    appBar?.navigationBar.adjustsFontSizeToFitWidth = true
```

Comment:
The variable you’re creating in line 581 should be lowercase, and it shouldn’t have the same name as its type. You can also get rid of the `self`. Something like this would be better:
`let appBarNavigationController = navigationController as? MDCAppBarNavigationController`
And maybe this is a matter of taste/preference, but in general, I think it’s best to minimize optional chaining. I think that’s what it’s called… I’m talking about this:
`let optional3 = optional1?.optional2?.optional3`
I would just replace these three lines with:
```Swift
    guard let appBarNavigationController = navigationController as? MDCAppBarNavigationController,
      let appBar = appBarNavigationController.appBarViewController(for: vc)
      else { return /* this is a good place to handle errors etc */ }
    appBar.navigationBar.adjustsFontSizeToFitWidth = true
```